pkg/config/setup/config.go

```diff
@@ -241,7 +243,16 @@ var serverlessConfigComponents = []func(pkgconfigmodel.Setup){
 func init() {
 	osinit()
 	// Configure Datadog global configuration
-	datadog = pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	if val, found := os.LookupEnv("DD_CONF_NODETREEMODEL"); found && val == "enable" {
+		datadog = nodetreemodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	} else if val, found := os.LookupEnv("DD_CONF_NODETREEMODEL"); found && val == "tee" {
+		var viperConfig = pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+		var nodetreeConfig = nodetreemodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+		datadog = teeconfig.NewTeeConfig(viperConfig, nodetreeConfig)
+	} else {
+		datadog = pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	}
+
```

Comment:
Nitpick: We could simplify this block to only look for the `DD_CONF_NODETREEMODEL` just once 😄 

```go
val, found := os.LookupEnv("DD_CONF_NODETREEMODEL")

if found  {
	if val == "enable" {
		datadog = nodetreemodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
	} else if val == "tee" {
		var viperConfig = pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
		var nodetreeConfig = nodetreemodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
		datadog = teeconfig.NewTeeConfig(viperConfig, nodetreeConfig)
	}
} else {
	datadog = pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
}
```